src/app/services/xml-parsers/apparatus-entries-parser.service.ts

```diff
@@ -15,4 +15,31 @@ export class ApparatusEntriesParserService {
     return Array.from(document.querySelectorAll<XMLElement>(this.appEntryTagName))
       .map((appEntry) => this.appParser.parse(appEntry));
   }
+
+  public getSignificantReadings(apps: ApparatusEntry[]) {
+    const signRdgs = {};
+    apps.forEach((app) => {
+      signRdgs[app.id] = app.readings.concat(app.lemma).filter((rdg: Reading) => rdg.significant);
+    });
+
+    return signRdgs;
+  }
+
+  public getSignificantReadingsNumber(signRdgs: { [key: string]: ApparatusEntry[] }) {
+    const signRdgsNumber = {};
+    Object.keys(signRdgs).forEach((app) => {
+      signRdgsNumber[app] = signRdgs[app].length;
+    });
+
+    return signRdgsNumber;
+  }
+
+  public getAppVariance(signRdgsNum: { [key: string]: number }, witList: { [key: string]: Witness }) {
+    const appsVariance = {};
+    Object.keys(signRdgsNum).forEach((x) => {
+      appsVariance[x] = signRdgsNum[x] / Object.keys(witList).length;
```

Comment:
Can `witList` have zero length?

Reply:
I don’t think so, but I'm not 100% sure. I'll find out and let you know.

Reply:
It's possible that there is no listWit in case it's a single-witness edition. In this case the variance value is useless.

Reply:
Ok for the usefulness, but what will happen to this snippet of code in such case?

Reply:
Returns an invalid value. I'm fixing it. :)